app/controllers/stages_controller.rb

```diff
@@ -142,6 +141,9 @@ def stage_permitted_params
         deploy_group_ids: [],
         command_ids: []
       }
-    ] + Samson::Hooks.fire(:stage_permitted_params).flatten
+    ].tap do |params|
+      params << :no_code_deployed if current_user.admin?
+      params << Samson::Hooks.fire(:stage_permitted_params)
+    end.flatten
```

Comment:
I don't think we want to flatten the entire param array... move this up to L#146?

Reply:
we are pushing into array in L#146? and so it will create a nested array, that's the reason we do flatten for entire params.

Reply:
I think only the result from `:stage_permitted_params` needs the flatten here..

Reply:
`samson(dev)> [9]+[8]
=> [9, 8]
samson(dev)> [9]<<[8]
=> [9, [8]]
`

Reply:
Then why not use `concat` and avoid the flatten?